src/game/scenes.ts

```diff
@@ -47,12 +51,37 @@ export function go(name: SceneName, ...args: unknown[]) {
             transform: new Mat23(),
         };
 
+        // clear the arguments //
+        _k.game.currentSceneArgs.splice(0, _k.game.currentSceneArgs.length);
+
+        _k.game.currentSceneArgs = args;
+
         _k.game.scenes[name](...args);
     });
 
     _k.game.currentScene = name;
 }
 
+export function pushScene(id: SceneName, ...args: unknown[])
+{
+    _k.game.sceneStack.push({
+        sceneID: _k.game.currentScene,
+        args: _k.game.currentSceneArgs,
+    });
+    go(id, args);
+    return;
+}
+
+export function popScene()
+{
+    const sceneData: SceneState | undefined  = _k.game.sceneStack.pop();
+    console.log(sceneData);
+    if (sceneData === undefined)
+        throw new Error("No more scenes to pop!");
+
+    go(sceneData.sceneID ?? "", sceneData.args);
```

Comment:
When is scene id null? At the start? Because if there is no concept of a default scene, go("") will fail.